src/utils/clickHandler.js

```diff
@@ -92,14 +94,17 @@ module.exports = function clickHandler (e) {
         }
     }
 
-    self.executeI13nEvent('click', {i13nNode: self.getI13nNode(), e:e}, function clickBeaconCallback () {
-        if (isRedirectLink) {
-            if (isFormSubmit(target)) {
-                // if the button has no form linked, then do nothing
-                target.form && target.form.submit();
-            } else {
-                document.location.assign(href);
+    // set-immediate to get click event fired with next tick, in case we need to update something before sending beacons
+    setImmediate(function executeI13nEvent() {
```

Comment:
nice catch, oops, I'm exactly thinking about the same scenario, but I suppose that we want to send `on` state with the beacon.

if we want `off`, we should keep the original implementation :)